### DISCUSSION TITLE
Is the HeadersAttribute affected by GHSA-3hxg-fxwm-8gf7?

### DISCUSSION THREAD

**Discussion by pkunze** [reactiveui/refit#1913] [Q&A] [answered]
Hi,

https://github.com/reactiveui/refit/security/advisories/GHSA-3hxg-fxwm-8gf7 does not exactly mention the `[Headers]` Attribute, but the [HeaderCollection]. Is it correct, that the [Headers]-Attribute ist not affected by the vulnerability?

TIA!

**Comment by glennawatson** [comment, marked as answer]
Correct - the `[Headers]` attribute is not the vector for GHSA-3hxg-fxwm-8gf7.

The vulnerability is CRLF injection through attacker-controlled runtime values, i.e. dynamic header values supplied via `[Header]` parameters and `[HeaderCollection]`. `[Headers("Name: value")]` values are compile-time string literals baked into the interface, so they are not attacker-controlled and were never the injection path.

That said, the fix in 7.2.22 / 8.0.0 is a single chokepoint: every header that Refit sets goes through one `SetHeader` method that strips CR and LF before calling `TryAddWithoutValidation`. So `[Headers]`, `[Header]`, and `[HeaderCollection]` are all sanitized regardless. You are safe on a patched version either way.
